app/controllers/dradis/plugins/csv/upload_controller.rb

```diff
@@ -2,24 +2,51 @@ module Dradis::Plugins::CSV
   class UploadController < ::AuthenticatedController
     include ProjectScoped
 
-    before_action :load_attachment, only: [:new]
+    before_action :load_attachment, only: [:new, :create]
 
     def new
       @default_columns = ['Unique Identifier', 'Column Header From File', 'Type', 'Field in Dradis']
 
       @headers = ::CSV.open(@attachment.fullpath, &:readline)
+
+      @log_uid = Log.new.uid + 1
     end
 
     def create
-      redirect_to main_app.project_upload_manager_path(current_project)
+      job_logger.write 'Enqueueing job to start in the background.'
+
+      MappingImportJob.perform_later(
+        file: @attachment.fullpath.to_s,
+        id_index: mappings_params[:identifier],
+        mappings: mappings_params[:mappings].to_h,
+        project_id: current_project.id,
+        uid: params[:log_uid].to_i
+      )
+
+      Resque.redis.del(params[:job_id])
+
+      head :ok
```

Comment:
What about rendering a create.js.erb here like in the UploadController where it initiates the ConsoleUpdater? That way we free up some code in the ever-growing upload.js

Reply:
Agree, moved to `create.js.erb`.